File: Python/Lineares/InsertionSort.py
def InsertionSort(vetor):
	for x in range(1, len(vetor)):
		chave = vetor[x]
		aux = x-1
		while aux >= 0 and vetor[aux] > chave:
			vetor[aux+1] = vetor[aux]
			aux -= 1
		vetor[aux+1] = chave
	return vetor

vetor = [92, 23, 42, 12, 54, 65, 1, 2, 8, 9, 31, 99]
print InsertionSort(vetor)

File: src/rust/insertion_sort.rs
fn main(){
  println!{"{:?}", insertion_sort(vec![54,42,11,33,24,99,77,80])};
}
fn insertion_sort(mut vetor: Vec<i32>) -> Vec<i32> {
  for i in 1..vetor.len() {
    let mut index: i32 = i as i32;
    while index > 0 && vetor[index as usize] < vetor[index as usize - 1] {
      vetor.swap(index as usize, index as usize - 1);
      index -= 1;
    }
  }
  vetor
}
